src/registry.rs

```diff
@@ -90,8 +91,43 @@ impl RegistryCore {
         }
 
         // TODO: metric_family injection hook.
-        // TODO: sort metrics.
-        mf_by_name.into_iter().map(|(_, m)| m).collect()
+
+        // Now that MetricFamilies are all set, sort their Metrics
+        // lexicographically by their label values.
+        for (_, ref mut mf) in &mut mf_by_name {
+            mf.mut_metric().sort_by(|&ref m1, &ref m2| {
+                let lps1 = m1.get_label();
+                let lps2 = m2.get_label();
+
+                if lps1.len() != lps2.len() {
+                    // This should not happen. The metrics are
+                    // inconsistent. However, we have to deal with the fact, as
+                    // people might use custom collectors or metric family injection
+                    // to create inconsistent metrics. So let's simply compare the
+                    // number of labels in this case. That will still yield
+                    // reproducible sorting.
+                    return lps1.len().cmp(&lps2.len());
+                }
+
+                for (lp1, lp2) in lps1.iter().zip(lps2.iter()) {
+                    if lp1.get_value() != lp2.get_value() {
+                        return lp1.get_value().cmp(lp2.get_value());
+                    }
+                }
+
+                // We should never arrive here. Multiple metrics with the same
+                // label set in the same scrape will lead to undefined ingestion
+                // behavior. However, as above, we have to provide stable sorting
+                // here, even for inconsistent metrics. So sort equal metrics
+                // by their timestamp, with missing timestamps (implying "now")
+                // coming last.
+                m1.get_timestamp_ms().cmp(&m2.get_timestamp_ms())
```

Comment:
Ported from client_golang, that is the way client_golang choose, rust should be the same, as it is ported from client_golang.